### DISCUSSION TITLE
Access components files using Monolith dashboard

### DISCUSSION THREAD

**Discussion by StefanoBerni** [elsa-workflows/elsa-core#2828] [Q&A]
Hi everyone,
I'd like to know how I can access (and modify) the files that provides the HTML component to the Elsa editor.
In the browser they are in the following path "/_content/Elsa.Designer.Components.Web/elsa-workflows-studio" but i can't find it in the project. Could anyone help me?
Thanks
Stefano

**Comment by sfmskywalker** [comment]
Hi @StefanoBerni 

The only way to change these files would be to clone the project and change the source component files that you find [here](https://github.com/elsa-workflows/elsa-core/tree/master/src/designer/elsa-workflows-studio).

**Reply by StefanoBerni** [reply]
Thanks, I've tried to modify that files but it doesn't happen anything. 
I've also tried to delete those files and I've discovered that they don't affect the application. Now in my project i need to add only few projects (imported as nuget packages), theese are: Elsa (with dependences), Elsa.Server.Api and Elsa.Persistence.EntityFramework.Core.
So, how can I modify the layout of the editor (I'm not using the dash at the moment).

**Reply by sfmskywalker** [reply]
When making changes to the dashboard/designer components, you need to also build them and copy the build output to your ASP.NET application into (a sub-folder of) the wwwroot folder so that these assets become available to the browser.

Then make sure that your HTML/Razor pages (e.g. _Host.cshtml) include the proper JS & CSS files now present in the wwwroot folder.

**Reply by StefanoBerni** [reply]
My _host.cshtml file calls some files from the directory "_content/Elsa.Designer.Component.Web/..." but i can't find this path in the project.
How can I do?

**Reply by sfmskywalker** [reply]
The `Elsa.Designer.Component.Web` package should have an embedded wwwroot folder containing the build output. You cannot edit these files, since they are embedded in this package.

Instead, you would need to make changes to the source files (in the `src/designer/elsa-workflows-studio` path), build that, copy the build output to your application, and update your _Host.cshtml file to point to that.

The `_content/Elsa.Designer.Component.Web/...` path would be of no use to you when you have a custom stencil build.

**Reply by SSCDanexis** [reply]
Hello i tried accessing the Elsa Root files to make some minor changes like adding some css classes and changing the Link Names (not the routes). 

After changing everything i used npm build in the /src/designer/elsa-workflows-root
Aswell as in the /src/designer/bindings/aspnet/Elsa.Designer.Components.Web folder and copied the wwwroot Output from that into my Published app. 

I also adjusted the _Host.cshtml File to point to those new Files, the changes applied, but now im not able to go into the workflow-registry link anymore, it just doesnt load and blocks every other link i click from that point out (until i refresh the website). 

Am i maybe missing something, what causes this? it also happens when i just build the default files and copy them. 

Browser Console shows the following: 

![image](https://user-images.githubusercontent.com/92802930/167661100-cf46f332-01c7-4f5a-bbde-742da595ff81.png)

**Reply by sfmskywalker** [reply]
The server responded with HTTP 400 - can you take a look at the actual HTTP request and its response?

**Reply by SSCDanexis** [reply]
i found a fix for this. 
The requested URL used to be https://localhost:5001/v1/workflow-registry/**by-provider**/ProgrammaticWorkflowProvider?version=Latest&page=0&pageSize=5

By deleting the "by-provider" Part in the elsa-client.ts it now works.
